01-basics/03-MarkedList/index.html

```diff
@@ -1,30 +1,30 @@
 <!DOCTYPE html>
 <html lang="ru">
-  <head>
-    <meta charset="UTF-8" />
-    <link rel="stylesheet" href="/assets/styles/index.css" />
-    <title>MarkedList</title>
-    <style>
-      .marked {
-        color: #efc14e;
-      }
-    </style>
-  </head>
-  <body>
-    <div id="app" style="margin-top: 2rem">
-      <div class="container">
-        <div class="form-group">
-          <input type="search" />
-        </div>
-        <ul>
-          <li>Email</li>
-          <li class="marked">Email</li>
-          <li>...</li>
-          <li>Email</li>
-        </ul>
+
+<head>
+  <meta charset="UTF-8" />
+  <link rel="stylesheet" href="/assets/styles/index.css" />
+  <title>MarkedList</title>
+  <style>
+    .marked {
+      color: #efc14e;
+    }
+  </style>
+</head>
+
+<body>
+  <div id="app" style="margin-top: 2rem">
+    <div class="container">
+      <div class="form-group">
+        <input v-model="search" type="search" />
       </div>
+      <ul>
+        <li :class="{marked: filteredEmails.includes(email)}" v-for="email in emails">{{ email }}</li>
```

Comment:
In the current implementation for rendering the list of emails, it iterates over all (filtered) emails. Before rendering, it loops over all emails to create the `filteredEmails` list, already checking each for compliance with the condition. Then on rendering each email, it loops over this list again with `.filter()` to check if this email is in the list of already checked emails. And the last operation happens each re-render even if no emails or search query was changed.

As a result, this `computed` doesn't help to render a marked emails list. It makes double work, making it in quadratic complexity, with calling a data processing method during the rendering.

Reply:
Agree with the comment. isMarked property was added to computed filtered array (for binding marked class to it). Also emails array moved from reactive data() to option of a component (as a constant).